### PR TITLE
add section: File resolution (#434)

### LINKED ISSUE #434: how does mach-nix resolve urls and hashes
there is [mach-nix#how-does-mach-nix-work](https://github.com/DavHau/mach-nix#how-does-mach-nix-work)

> As core for the resolving resolvelib is used: https://github.com/sarugaku/resolvelib
>
> Mach-nix supports multiple providers to retrieve python packages from. The user can specify which providers should be preferred. Packages from different providers can be mixed.

but it does not explain how mach-nix resolves urls and hashes of the sdist/wheel/conda files

from the nix point of view, that is the magic part
(assuming that `fetchurl` is only allowed if we know the hash before downloading)

### PR DESCRIPTION
**Pull request by milahu**
fix #434

maybe a bit too verbose ...

maybe belongs to `eli5.md` ; )

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -203,6 +204,60 @@ As core for the resolving resolvelib is used: https://github.com/sarugaku/resolv
...
+---------  ---------- -----   ----
+        0  04-01-2022 10:08   PIL/
+        0  04-01-2022 10:08   Pillow.libs/
+        0  04-01-2022 10:08   Pillow-9.1.0.dist-info/
+     7311  04-01-2022 10:08   PIL/PdfImagePlugin.py
+   141328  04-01-2022 10:08   PIL/_imagingcms.cpython-39-x86_64-linux-gnu.so
+    46872  04-01-2022 10:08   PIL/_imagingtk.cpython-39-x86_64-linux-gnu.so
+     1513  04-01-2022 10:08   PIL/GribStubImagePlugin.py
+```
+
```
Comment (DavHau) [inline Readme.md:260]: Is this really relevant to the user?
Should this better go to a contributors guide?

Comment (milahu) [reply Readme.md:260]: yes ... lets move the whole `how-does-mach-nix-work` section to `implementation.md`?
